R/modify-model-field.R

```diff
@@ -85,6 +85,8 @@ replace_model_field <- function(.mod, .field, .old_val, .new_val) {
 
   .mod[[.field]][idx] <- .new_val
 
+  .mod <- save_model_yaml(.mod)
```

Comment:
This change adds a side effect to `replace_model_field()`. This is consistent with the documentation for `modify_model_field()`

> After the object has been modified they will write the modified object back to the YAML

but worth commenting on because in general we should try to isolate side effects to as few functions as possible. This may not be achievable unless we change the abstraction so that the model is entirely in memory (versus portions in both memory and on disk).

Reply:
:eye_roll: You know I don't disagree, but you also know that ain't happening. (the "entirely in memory" comment)